### ISSUE TITLE
Restrict IP access

### ISSUE THREAD

**Issue by omahonyb** [neutrinolabs/xrdp#1195]
I am wondering is there a way to restrict the incoming access to xrdp via RDP to a specific IP or range? My setup is working, but I don't want it open to everyone on the network segment that can access the IP.

**Comment by metalefty** [comment]
Do it in your OS firewall.

**Comment by omahonyb** [comment]
Yeah I was hoping there was an easier way. I don't want to turn on iptables for one service, but thanks anyway.

**Comment by ghost** [comment]
Id like to mention the persons wanting to increase security should consider using SSH tunneling. With openSSH server its enabled by default and no configuration is necessary.

Just open Remmina for example and click "Enable SSH Tunnel", I then use "Public Key Automatic" as my keys are already stored. Make sure to select "Tunnel via loopback address" or it will connect via the hosts interface address rather than 127.0.0.1

You can modify the port entry in /etc/xrdp/xrdp.ini to only allow inbound RDP connections from the SSH tunnel itself by setting

port=tcp://127.0.0.1:3389

SSH eliminates both password attacks and RDP authentication vulnerabilities as well as potential weaknesses in the encryption of the RDP stream itself. I personally would never use RDP without SSH if the LAN was not in a controlled environment, Wifi was present or the LAN was internet accessible on port 3389. SSH is preferable to VPN or at the very least should be used in addition to VPN. VPN networks, particularly where lan-to-lan VPN is used can have gaping vulnerabilities, especially if those LANs have WiFi. SSH provides end-to-end encryption and authentication, that's quite a bit better than VPN in most cases anyway.

Other advantages of using SSH tunneling for xrdp would be the use of fail2ban, logging, auditing and the other benefits of SSH in general. The ability to choose the type and level of encryption you want like AES-256-GCM, hmac-sha2-512, ecdh-sha2-nistp521 etc. as well as the strong anti man-in-the-middle abilities of SSH.

One final point is that SSH can safely connect to an xrdp host over the internet without even the need of VPN; SSH is completely secure all by itself with a standard hardened configuration. This is a very attractive alternative considering VPN could be overkill, difficult to setup at first and still not even be as secure as just using SSH alone.

Typical SSH hardening config would be:
PasswordAuthentication no
PermitRootLogin no
MaxAuthTries 3
Protocol 2
Ciphers [aes256-gcm@openssh.com](mailto:aes256-gcm@openssh.com)
KexAlgorithms ecdh-sha2-nistp521
MACs [hmac-sha2-512-etm@openssh.com](mailto:hmac-sha2-512-etm@openssh.com)
HostKey /etc/ssh/ssh_host_rsa_key

and generate both a strong server key pair and user key pair:
sudo ssh-keygen -t rsa-sha2-512 -b 16384 -N '' -f /etc/ssh/ssh_host_rsa_key <<<y
ssh-keygen -N '' -t rsa -b 16384
